atomate/vasp/workflows/base/magnetism.py

```diff
@@ -0,0 +1,834 @@
+        # now done by a full symmetry analysis
+        sga = SpacegroupAnalyzer(structure)
+        structure_sym = sga.get_symmetrized_structure()
+        wyckoff = ["n/a"] * len(structure)
+        for indices, symbol in zip(
+            structure_sym.equivalent_indices, structure_sym.wyckoff_symbols
+        ):
+            for index in indices:
+                wyckoff[index] = symbol
+        is_magnetic_sites = [
+            True if site.specie in types_mag_species else False for site in structure
+        ]
+        # we're not interested in sites that we don't think are magnetic,
+        # set these symbols to None to filter them out later
+        wyckoff = [
+            symbol if is_magnetic_site else "n/a"
+            for symbol, is_magnetic_site in zip(wyckoff, is_magnetic_sites)
+        ]
+        structure.add_site_property("wyckoff", wyckoff)
+        wyckoff_symbols = set(wyckoff) - {"n/a"}
+
+        # if user doesn't specifically request ferrimagnetic_Cr2NiO4 orderings,
+        # we apply a heuristic as to whether to attempt them or not
+        if self.automatic:
+            if (
+                "ferrimagnetic_by_motif" not in self.strategies
+                and len(wyckoff_symbols) > 1
+                and len(types_mag_species) == 1
+            ):
+                self.strategies += ("ferrimagnetic_by_motif",)
+
+            if (
+                "antiferromagnetic_by_motif" not in self.strategies
+                and len(wyckoff_symbols) > 1
+                and len(types_mag_species) == 1
+            ):
+                self.strategies += ("antiferromagnetic_by_motif",)
+
+            if (
+                "ferrimagnetic_by_species" not in self.strategies
+                and len(types_mag_species) > 1
+            ):
+                self.strategies += ("ferrimagnetic_by_species",)
+
+        # we start with a ferromagnetic ordering
+        if "ferromagnetic" in self.strategies:
+
+            # TODO: remove 0 spins !
+
+            fm_structure = analyzer.get_ferromagnetic_structure()
+            # store magmom as spin property, to be consistent with output from
+            # other transformations
+            fm_structure.add_spin_by_site(fm_structure.site_properties["magmom"])
+            fm_structure.remove_site_property("magmom")
+
+            # we now have our first magnetic ordering...
+            self.ordered_structures.append(fm_structure)
+            self.ordered_structure_origins.append("fm")
+
+        # we store constraint(s) for each strategy first,
+        # and then use each to perform a transformation later
+        all_constraints = {}
+
+        # ...to which we can add simple AFM cases first...
+        if "antiferromagnetic" in self.strategies:
+
+            constraint = MagOrderParameterConstraint(
+                0.5,
+                # TODO: this list(map(str...)) can probably be removed
+                species_constraints=list(map(str, types_mag_species)),
```

Comment:
can you remove list(map(str ?

If you leave it as a comment, no one is going to ever touch it ...

Reply:
Noted 👍